state/protocol/badger/mutator.go

```diff
@@ -556,6 +556,14 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 
 	// track service event driven metrics and protocol events that should be emitted
 	var events []func()
+
+	// track service events to index
+	var serviceEvents []struct {
```

Comment:
This type is defined quite far away from where it's used. Better to move it close.

Reply:
https://github.com/onflow/flow-go/pull/3605#discussion_r1029935694 fixed this as well